Socket connection fails if space is present after URL

The block will fail to connect and dump an error when there is a space after the URL in the "SocketIO host" field. The error it dumps does not give indication that there is an issue in syntax, but instead looks like this: 

`[2016-02-17 20:09:08.633] NIO [ERROR] [Socket Test.Socket] Error connecting, (Exception details: ConnectionError - ('Connection aborted.', gaierror(8, 'nodename nor servname provided, or not known')))`

--- a/socketio_block.py
+++ b/socketio_block.py
@@ -224,7 +224,7 @@ class SocketIO(Retry, Block):
         self._client.connect()
 
     def _build_socket_url_base(self):
-        host = self.host()
+        host = self.host().strip()
         # Default to http protocol
         # See if they included an http or https in front of the host,
         host_matched = re.match('^(https?)://(.*)$', host)

--- a/tests/test_socketio_block.py
+++ b/tests/test_socketio_block.py
@@ -289,3 +289,26 @@ class TestSocketIOBlock(NIOBlockTestCase):
         parent_log = super().get_logging_config()
         parent_log['handlers']['default']['level'] = 'INFO'
         return parent_log
+
+    def test_host_name_spaces(self):
+        """ Test that an error isn't thrown when there is a space after the
+        hostname.
+        """
+
+        blk = SocketIO()
+        blk._do_handshake = MagicMock()
+        blk._close_client = MagicMock()
+
+        # this will raise a socket.gaierror if the hostname has a space at the
+        # end/ an invalid host is passed to socket.getaddrinfo(). If the block
+        # is given a valid hostname, it will throw a ConnectionRefusedError
+        # on this test.
+        with self.assertRaises(ConnectionRefusedError):
+            self.configure_block(blk, {
+                "host": "127.0.0.1 ",
+                "port": 80,
+                "room": "myroom",
+                "listen": True,
+                "start_without_server": False,
+            })
+        self.assertFalse(blk._client_ready)

